src/Autocomplete/Domain/DataAccess/DoctrineLocationRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Autocomplete\Domain\DataAccess;
+
+use Doctrine\ORM\EntityManager;
+use WMDE\Fundraising\Frontend\Autocomplete\Domain\LocationRepository;
+use WMDE\Fundraising\Frontend\Autocomplete\Domain\Model\Location;
+
+class DoctrineLocationRepository implements LocationRepository {
+
+	private EntityManager $entityManager;
+
+	public function __construct( EntityManager $entityManager ) {
+		$this->entityManager = $entityManager;
+	}
+
+	public function getCitiesForPostcode( string $postcode ): array {
+		$queryBuilder = $this->entityManager->getConnection()->createQueryBuilder();
+		$metaData = $this->entityManager->getClassMetadata( Location::class );
+
+		$districtField = $metaData->getColumnName( 'districtName' );
```

Comment:
In this set it actually also means district, eg putting in your Berlin postcode returns your city district

Reply:
Yeah, I now realize that the query might not be as straightforward, because `ORT_NAME` can either mean town name OR district name of big city. For big cities like Berlin, Munich Frankfurt, etc using `ORT_NAME` is useless. For small towns it's the only way to get to the town name.

Looking at the data, it might be that we can look at `GEMEINDE_AGS` and if that is `Stadt` we need to query `communityName` (`districtName` is still too broad), otherwise `cityName`. We'll ask Kai if there is some recommended/documented way to query for a City/Town name by the data provider.

Reply:
Isn't the value in `communityName` always correct? I just tried all the places in Germany I've ever lived and got the correct city name from that field. It also contains "Heimertingen" for the postal code 87751.

Reply:
`communityName` seems quite good, I can +1 that.

For example (place near where I used to live :grimacing: ) Gemeinde Beuron, there are 5 different actual locations (Orte) / tiny different villages, but they all share the same postcode anyway. 
When I look up some venue addresses there all use "88631 Beuron" instead of e.g. "88631 Thiergarten".
Some of them use "88631 Beuron - Thiergarten" or "88631 Beuron / Thiergarten", but Deutsche Post should be able to handle "Beuron" only, right

Reply:
Nice, I didn't notice that. I'll update on Monday